### ISSUE TITLE
Failure tries to extract a traceback from non-traceback objects on Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8562]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8562|
|Type|enhancement|
|Created|2016-07-05 14:35:42Z|
|Branch|https://github.com/twisted/twisted/tree/failure-py3-8562|

```
[ERROR]
Traceback (most recent call last):
  File "/Users/red/code/twgit/twisted/internet/_baseprocess.py", line 64, in maybeCallProcessEnded
    proto.processEnded(Failure(reason))
  File "/Users/red/code/twgit/twisted/internet/utils.py", line 166, in processEnded
    self.deferred.errback((out, err, e.signal))
  File "/Users/red/code/twgit/twisted/internet/defer.py", line 433, in errback
    fail = failure.Failure(fail)
  File "/Users/red/code/twgit/twisted/python/failure.py", line 232, in __init__
    tb = self.value.__traceback__
builtins.AttributeError: 'tuple' object has no attribute '__traceback__'

twisted.test.test_iutils.ProcessUtilsTests.test_outputSignal
```

<details><summary>Searchable metadata</summary>

```
trac-id__8562 8562
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__Python_3_x Python-3.x
branch__failure_py3_8562 failure-py3-8562
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1467729342693353 1467729342693353
changetime__1501847537684117 1501847537684117
version__None None
owner__None None

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

So with talking with Tom Prince, this is a bug that shouldn't be fixed (but the ```getProcess*``` APIs that rely on it should be fixed instead). So, we should give a proper reason for this not working (raising an exception when trying to errback on Python 3 with arbitrary types, maybe).

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc commented|
|-|-|

Fixed in In [changeset:"f2e0b9f584c8508241ff15b3898c3b208b4d6d2e" f2e0b]

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)|@rodrigc set status to `closed`|
|-|-|
